src/org/traccar/database/DataManager.java

```diff
@@ -501,224 +268,52 @@ public void addStatistics(Statistics statistics) throws SQLException {
                 .executeUpdate());
     }
 
-    public Collection<Calendar> getCalendars() throws SQLException {
-        return QueryBuilder.create(dataSource, getQuery("database.selectCalendarsAll"))
-                .executeQuery(Calendar.class);
+    public static String makeName(String object) {
+        return object.substring(0, 1).toUpperCase() + object.replace("Id", "").substring(1);
     }
 
-    public void addCalendar(Calendar calendar) throws SQLException {
-        calendar.setId(QueryBuilder.create(dataSource, getQuery("database.insertCalendar"), true)
-                .setObject(calendar)
-                .executeUpdate());
-    }
-
-    public void updateCalendar(Calendar calendar) throws SQLException {
-        QueryBuilder.create(dataSource, getQuery("database.updateCalendar"))
-                .setObject(calendar)
-                .executeUpdate();
-    }
-
-    public void removeCalendar(long calendarId) throws SQLException {
-        QueryBuilder.create(dataSource, getQuery("database.deleteCalendar"))
-                .setLong("id", calendarId)
-                .executeUpdate();
+    public static String makeNameId(String object) {
+        return object.substring(0, 1).toLowerCase() + object.substring(1) + (object.indexOf("Id") == -1 ? "Id" : "");
     }
 
-    public Collection<CalendarPermission> getCalendarPermissions() throws SQLException {
-        return QueryBuilder.create(dataSource, getQuery("database.selectCalendarPermissions"))
-                .executeQuery(CalendarPermission.class);
-    }
+    public void linkObject(String owner, long ownerId, String property, long propertyId,
+            boolean link) throws SQLException {
+        String query = "database." + (!link ? "un" : "") + "link" + makeName(owner) + makeName(property);
```

Comment:
I think it would be better to use `(link ? "link" : "unlink")` here. It's shorter and much more readable.